pkg/pquerier/collector.go

```diff
@@ -263,3 +273,67 @@ func aggregateClientAggregatesCrossSeries(ctx *selectQueryContext, res *qryResul
 	lastT, lastV := it.At()
 	return lastT, lastV, nil
 }
+
+func intervalAggregation(ctx *selectQueryContext, res *qryResults, t int64, v float64) {
+	currentCell := (t - ctx.queryParams.From) / res.query.aggregationParams.Interval
+
+	for _, col := range res.frame.columns {
+		if col.GetColumnSpec().metric == res.name {
+			_ = res.frame.setDataAt(col.Name(), int(currentCell), v)
+		}
+	}
+}
+
+func windowAggregation(ctx *selectQueryContext, res *qryResults, t int64, v float64) {
+	currentCell := (t - ctx.queryParams.From) / res.query.aggregationParams.Interval
+	if (t-ctx.queryParams.From)%res.query.aggregationParams.Interval > 0 {
+		currentCell++
+	}
+
+	if res.query.aggregationParams.GetAggregationWindow() > res.query.aggregationParams.Interval {
+		currentCellTime := ctx.queryParams.From + currentCell*res.query.aggregationParams.Interval
+		maximumAffectedTime := t + res.query.aggregationParams.GetAggregationWindow()
+		numAffectedCells := (maximumAffectedTime-currentCellTime)/res.query.aggregationParams.Interval + 1 // +1 to include the current cell
+
+		for i := int64(0); i < numAffectedCells; i++ {
```

Comment:
why to cast here? cast `numAffectedCells` to int and that's it. Right?

Reply:
then i would need to also cast `currentCell` to int anyway so it doesn't save a cast